.github/CONTRIBUTING.md

```diff
@@ -186,23 +186,23 @@ Signed-off-by: Contributor <contributor@email.com>
 ```
 
 #### OpenZFS Patch Ports
-If you are porting an OpenZFS patch, the commit message must meet
+If you are porting OpenZFS patches, the commit message must meet
 the following guidelines:
-* The first line must be the summary line from the OpenZFS commit.
-It must begin with `OpenZFS dddd - ` where `dddd` is the OpenZFS issue number.
-* Provides a `Authored by:` line to attribute the patch to the original author.
-* Provides the `Reviewed by:` and `Approved by:` lines from the original
+* The first line must be the summary line from the OpenZFS commit(s).
+It must begin with `OpenZFS dddd, dddd - ` where `dddd` are OpenZFS issue numbers.
```

Comment:
The first line is specified elsewhere in the `CONTRIBUTING` document as being "50 characters or less". It's not realistic to include multiple OpenZFS numbers and multiple summary lines all within the mandated 50 chars. Even a single OpenZFS commit number + summary line will often exceed the 50 char limit. How should this be resolved?

Reply:
That requirement is intentionally left out because, as you already mentioned, OpenZFS ports will have subject lines longer than 50 characters. Because it isn’t explicitly stated, we do not impose the 50 character limit in the commit message checking.

Reply:
If the 50 char limit stated elsewhere doesn't apply in this case, that should be explicit here I think.

But with 2 or more OpenZFS patches being applied, "The first line must be the summary line from the OpenZFS commit(s)." indicates multiple OpenZFS summary lines should be concatenated into the ZoL summary line. That's not realistic.

As a concrete example, how should @behlendorf's recent commit message in #6779 be formatted?

Reply:
It should be `OpenZFS 8558, 8602 - lwp_create() returns EAGAIN`. That’s because 8602 is a bug fix for 8558.

Reply:
Ok, so you've used the summary line from first (or primary) OpenZFS commit rather than from multiple summary lines. Perhaps that should be the instruction? I.e.
```
The first line must be the summary line from first or most important OpenZFS commit.
```

On the other hand there's also 2a8b84b which commits two related but independent patches, i.e. neither is really "primary". Perhaps the solution there is to insist on separate patches except where the secondary patches are directly related to the first, e.g. bug fix or cleanup.

Reply:
Note: in the end I'm actually not too worried about all this so happy to have "whatever" committed.

Reply:
@chrisrd I can make a slight tweak on this at some point today. Thanks for the feedback!